### DISCUSSION TITLE
How to debug failed upgrade?

### DISCUSSION THREAD

**Discussion by xsuchy** [fwupd/fwupd#5360] [General]
Fwupdmgr offered me an upgrade of firmware (IIRC Intel machine code). I applied it and after reboot, I got an error. Something like "Cannot apply the capsule". And within few seconds the machine was rebooted.
I want to debug this issue and report it back. But I have no idea how. I searched the website, this discussion section. But still, no guidance on how to debug such failures.

**Comment by hughsie** [comment]
`fwupdmgr get-history` should show you what was attempted.

**Reply by hughsie** [reply]
-- adding to that; if the failure is recognised we show the user a specific wiki link, but that requires the user to submit a report back to the LVFS first.

**Comment by xsuchy** [comment]
```
$ LC_ALL=C fwupdmgr get-history
...
??Intel Management Engine:
? ?   Device ID:          349bb341230b1a86e5effe7dfe4337e1590227bd
? ?   Previous version:   192.90.2072
? ?   Update State:       Failed
? ?   Update Error:       failed to run update on reboot
? ?   Last modified:      2022-12-18 21:55
? ?   GUID:               2c0665e2-fdbd-495e-b8e4-69d92b9c119a
? ?   Device Flags:       ? Internal device
? ?                       ? Updatable
? ?                       ? System requires external power source
? ?                       ? Needs a reboot after installation
? ?                       ? Device is usable for the duration of the update
? ? 
? ??  New version:      192.92.2145
?       License:          Unknown
?       Description:      
?       The vendor did not supply any release notes.
```

I run `fwupdmgr report-history` and subsequent run of `get-history` does not show any link.

But my question was not about this specific update. I think, I am poweruser and have enough experience to dive into logs, configuration files etc. But without any supporting documentation, I am forced to spend lots of time digging in various resources and a code. So I am looking for something like:
* if something doesn't work, please run `fwupdmgr report-history`.
* if you want to dive into it more, please run this log...
* if you dive into it even more then...
* even more? check this...

**Reply by superm1** [reply]
In the case of UEFI based updates like this, we don't turn on the debug logs by default.  This is intentional to not waste the cycles on the NVRAM.  The next step is to use `fwupdate` to turn those on.

`fwupdate -d` will turn that flag on.  Run your update again (and expect it to fail).
Then after reboot you can run `fwupdate -L` to get the log that was created.

If that log shows the update was actually attempted but nothing happened there isn't anything more you can do; it's entirely up to the vendor in their BIOS implementation to handle the update from that hand-off onwards.
